internal/controller/serverset/bootvolume_controller_test.go

```diff
@@ -0,0 +1,104 @@
+package serverset
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+	_ "sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/apis/compute/v1alpha1"
+	computev1alpha1 "github.com/ionos-cloud/crossplane-provider-ionoscloud/apis/compute/v1alpha1"
+)
+
+func fakeKubeClient(functions interceptor.Funcs) client.WithWatch {
+	scheme, _ := computev1alpha1.SchemeBuilder.Build()
+	return fake.NewClientBuilder().WithScheme(scheme).WithInterceptorFuncs(functions).Build()
+}
+
+func getVolumePopulateStatusTest(ctx context.Context, client client.WithWatch, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+	vol := obj.(*v1alpha1.Volume)
+	vol.Status.AtProvider.State = "AVAILABLE"
+	vol.Status.AtProvider.VolumeID = "uuid"
+	return nil
+}
+
+func createVolumeReturnsErrorTest(ctx context.Context, client client.WithWatch, obj client.Object,
+	opts ...client.CreateOption) error {
+	return errors.New("something went wrong")
+}
+
+func Test_kubeBootVolumeController_Create(t *testing.T) {
+	type fields struct {
+		kube client.Client
+		log  logging.Logger
+	}
+	type args struct {
+		ctx          context.Context
+		cr           *v1alpha1.ServerSet
+		replicaIndex int
+		version      int
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    v1alpha1.Volume
+		wantErr bool
+	}{
+		{
+			name: "expect success",
```

Comment:
I would say to give more details about what this scenario is testing. This string is too general. I was thinking to something like: "WhenBootVolumeSuccessfullyCreatedThenVolumeStatusIsPopulated". From a test I expect to get an answer to the following questions:
- what does it testing? (e.g. boot volume creation)
- what scenario? (e.g. boot volume successfully created)
- what is the expected result? (e.g. volume status is populated)

Reply:
So when you run the test this is the output: 
=== RUN   Test_kubeBootVolumeController_Create
=== RUN   Test_kubeBootVolumeController_Create/expect_success
=== RUN   Test_kubeBootVolumeController_Create/expect_error
--- PASS: Test_kubeBootVolumeController_Create (0.00s)
    --- PASS: Test_kubeBootVolumeController_Create/expect_success (0.00s)
    --- PASS: Test_kubeBootVolumeController_Create/expect_error (0.00s)
If you want I can add some more info, but don't want to duplicate the `testbootvolumecreate` part.